libs/h3/src/main/java/org/elasticsearch/h3/HexRing.java

```diff
@@ -569,6 +569,24 @@ final class HexRing {
             CoordIJK.Direction.CENTER_DIGIT,
             CoordIJK.Direction.IJ_AXES_DIGIT } };
 
+    private static final CoordIJK.Direction[] NEIGHBORSETCLOCKWISE = new CoordIJK.Direction[] {
+        CoordIJK.Direction.CENTER_DIGIT,
+        CoordIJK.Direction.JK_AXES_DIGIT,
+        CoordIJK.Direction.IJ_AXES_DIGIT,
+        CoordIJK.Direction.J_AXES_DIGIT,
+        CoordIJK.Direction.IK_AXES_DIGIT,
+        CoordIJK.Direction.K_AXES_DIGIT,
+        CoordIJK.Direction.I_AXES_DIGIT };
+
+    private static final CoordIJK.Direction[] NEIGHBORSETCOUNTERCLOCKWISE = new CoordIJK.Direction[] {
```

Comment:
I don't quite understand the code where these are used, but cannot help notice that they do not represent the opposite order of each other. I see that `JK, IJ, J, IK, K, I` is not the reverse of `IK, JK, K, IJ, I, J`.

Reply:
This is coming from here: https://github.com/uber/h3/blob/87de60bb5c88a72b09c7c8dca0c2df4e7ac1ddef/src/h3lib/lib/directedEdge.c#L94

I am afraid if you really wan to understand it, you need to ask the creators of H3